src/test/java/org/spongepowered/common/registry/CatalogTypeMethodTest.java

```diff
@@ -138,31 +110,47 @@ private void testResult(Object object) {
                 }
             }
         }
-        if (object instanceof Translatable) {
-            testResult(((Translatable) object).getTranslation());
+
+        if (object instanceof SpongeCatalogType) {
+            String name = ((SpongeCatalogType) object).getName();
+            String id = ((SpongeCatalogType) object).getId();
+
+            if (name.equals(id)) {
+                LOG.warn("id's and names should NEVER be the same. Catalog Type: {} | Id : {} | Name : {} | ({})", this.name, id, name, this.implementationClass);
+            }
+
         }
-        if (object instanceof Translation) {
-            Translation translation = (Translation) object;
-            String translationId = checkNotNull(translation.getId(), "translationId");
-            String translated = checkNotNull(translation.get(), "translated");
-            if (translationId.equals(translated)) {
-                boolean ignore = false;
-                boolean silent = false;
-                for (Entry<String, Boolean> entry : ignoredTranslationPrefixes.entrySet()) {
-                    if (translationId.startsWith(entry.getKey())) {
-                        ignore = true;
-                        silent = entry.getValue();
-                    }
+
+        if (object instanceof Translatable) {
+            Translation translation = ((Translatable) object).getTranslation();
+            if (ignoredTranslationPrefixes.entrySet().stream().anyMatch(entry -> translation.getId().startsWith(entry.getKey()) && entry.getValue())) {
+                if (verbose) {
+                    LOG.info("Ignored {}", translation.getId());
                 }
-                if (silent) {
-                    // Do nothing
-                } else if (ignore) {
-                    LOG.warn("Catalog Type: {} Id : {} fails to provide translation: '{}' in {} ({})",
-                            this.name, this.catalogId, translationId, this.methodName, this.implementationClass);
-                } else {
-                    throw new RuntimeException("No translation present for " + translationId);
+                return;
+            }
+
+
+            if (object instanceof IStringSerializable || object instanceof Fish || object instanceof CookedFish) {
+                if (verbose) {
+                    LOG.info("This is probably a blockstate. Mojang doesn't directly provide translations for blockstates. Ignoring {}", translation.getId());
```

Comment:
Why is fish considered untranslatable but you're saying it's probably a block state?

Reply:
Indeed. However that's net.minecraft.item.ItemFishFood.

This check should avoid net.minecraft.item.ItemFishFood.FishType